src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -276,6 +276,12 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
             {
                 return false;
             }
+            else if (inputType.TypeKind == TypeKind.Pointer || patternType.TypeKind == TypeKind.Pointer)
```

Comment:
Are we testing the second condition: `patternType.TypeKind == TypeKind.Pointer`?

Reply:
@cston This is a kind of future-proofing. I don't know of a way a pointer type can arise as a pattern's type due to the structure of the language grammar, but if it does arise (due to some future kind of type alias, for example) we'd want this behavior.